test/integration/error.test.js

```diff
@@ -84,15 +84,102 @@ describe(Support.getTestDialectTeaser('Sequelize Errors'), () => {
     });
 
+    it('SequelizeValidationErrorItem does not require instance & validator constructor parameters', () => {
+      const error = new Sequelize.ValidationErrorItem('error!', null, 'myfield');
+
+      expect(error).to.be.instanceOf(Sequelize.ValidationErrorItem);
+    });
+
+    it('SequelizeValidationErrorItem should have instance, key & validator properties when given to constructor', () => {
+      const inst  = { foo : 'bar' };
+      const vargs = [4];
+
+      const error = new Sequelize.ValidationErrorItem('error!', 'FUNCTION', 'foo', 'bar', inst, 'klen', 'len', vargs);
+
+      expect(error).to.have.property('instance');
+      expect(error.instance).to.equal(inst);
+
+      expect(error).to.have.property('validatorKey',  'klen');
+      expect(error).to.have.property('validatorName', 'len');
+      expect(error).to.have.property('validatorArgs', vargs);
+    });
+
+    it('SequelizeValidationErrorItem.getValidatorKey() should return a string', () => {
+      const error = new Sequelize.ValidationErrorItem('error!', 'FUNCTION', 'foo', 'bar', null, 'klen', 'len', [4]);
+
+      expect(error).to.have.property('getValidatorKey');
+      expect(error.getValidatorKey).to.be.a('function');
+
+      expect(error.getValidatorKey())           .to.equal('function.klen');
+      expect(error.getValidatorKey(false))      .to.equal('klen');
+      expect(error.getValidatorKey(0))          .to.equal('klen');
+      expect(error.getValidatorKey(1, ':'))     .to.equal('function:klen');
+      expect(error.getValidatorKey(true, '-:-')).to.equal('function-:-klen');
+
+      const empty = new Sequelize.ValidationErrorItem('error!', 'FUNCTION', 'foo', 'bar');
+
+      expect(empty.getValidatorKey())           .to.equal('');
+      expect(empty.getValidatorKey(false))      .to.equal('');
+      expect(empty.getValidatorKey(0))          .to.equal('');
+      expect(empty.getValidatorKey(1, ':'))     .to.equal('');
+      expect(empty.getValidatorKey(true, '-:-')).to.equal('');
+    });
+
+    it('SequelizeValidationErrorItem.getValidatorKey() should throw if namespace separator is invalid (only if NS is used & available)', () => {
+      const error = new Sequelize.ValidationErrorItem('error!', 'FUNCTION', 'foo', 'bar', null, 'klen', 'len', [4]);
+
+      expect(() => error.getValidatorKey(false, {}))       .to.not.throw();
+      expect(() => error.getValidatorKey(false, []))       .to.not.throw();
+      expect(() => error.getValidatorKey(false, null))     .to.not.throw();
+      expect(() => error.getValidatorKey(false, ''))       .to.not.throw();
+      expect(() => error.getValidatorKey(false, false))    .to.not.throw();
+      expect(() => error.getValidatorKey(false, true))     .to.not.throw();
+      expect(() => error.getValidatorKey(false, undefined)).to.not.throw();
+      expect(() => error.getValidatorKey(true, undefined)) .to.not.throw(); // undefined will trigger use of function parameter default
```

Comment:
The test is, per description, for the error case. But `.to.not.throw` doesn't test an error case obviously. This should be covered by the test for the normal case.